arm11/source/start.s

```diff
@@ -127,4 +143,19 @@ wait_arm9:
 	@ Jump to the kernel!
 	bx lr
 
+smp_boot:
+	@ r0 is the current CPU ID
+	ldr r1, =SYNC_ADDR
+	add r1, r1, r0, lsl #2
+	mov r12, #0
+	str r12, [r1]
```

Comment:
Wouldn't it be better if this was cleared by CPU0 before it initializes (boots) the rest of the cores? This way we avoid a possible race condition.

Reply:
so far I haven't encountered any and this is mostly just a stopgap measure for now, until we have a fully fledged bootloader that can handle the extra cores

Reply:
I see @xerpi 's point, if cores 1, 2, and 3 were executing this code. Right now, only core 1 will be.  When we add support for cores 2, 3, then yeah we won't want them to reset *SYNC_ADDR on each other.

Reply:
Right now only core1 is bootable anyway, when adding support for CPU2 and CPU3 on new3DS we'll need a more robust set of functions to deal with this

Reply:
Also, because we shift the cpuid by 4 and add it to sync_addr, each core is looking at separate memory locations. So they're not all looking at *SYNC_ADDR.

Reply:
What I meant is that we have to make sure the core is already spinning in the `ldr` (so it has already `str` with 0 before) before we let Linux write the jump address, otherwise the `str` would replace it with a 0.